lib/protect.ex

```diff
@@ -56,71 +61,95 @@ defmodule Protect do
       raise "--rules must be a json file"
     end
 
-    Enum.each([options[:org], options[:user]], &(
-      if &1 && Regex.match?(~r/[^a-zA-Z0-9\-]/, &1) do
+    Enum.each(
+      [options[:org], options[:user]],
+      &if &1 && Regex.match?(~r/[^a-zA-Z0-9\-]/, &1) do
         raise "user/org must be a valid Github username"
       end
-    ))
+    )
 
     options
   end
 
   @doc """
     Gets a list of Github repos for an owner that can be either a user or an
-    organisation.
+    organization.
   """
   def get_repos(options, page \\ 1, repos \\ []) do
     new_repos =
       options
       |> get_repos_url(page)
-      |> Github.get!
+      |> Github.get!()
       |> Map.get(:body, "{}")
-      |> Poison.decode!
+      |> Poison.decode!()
       |> Enum.map(&Map.fetch!(&1, "name"))
 
     case length(new_repos) < 100 do
       true -> repos ++ new_repos
-      _ -> get_repos options, page + 1, repos ++ new_repos
+      _ -> get_repos(options, page + 1, repos ++ new_repos)
+    end
+  end
+
+  defp owner(options) do
+    cond do
+      options[:org] ->
+        options[:org]
+
+      options[:user] ->
+        options[:user]
     end
   end
 
   @doc """
     Determines the correct url to use in the get_repos function, depending on
-    whether the repo owner is a user or an organisation.
+    whether the repo owner is a user or an organization.
   """
   def get_repos_url(options, page \\ 1) do
-      cond do
-        options[:org] ->
-          "/orgs/#{options[:org]}/repos?per_page=100&page=#{page}"
-        options[:user] ->
-          "/users/#{options[:user]}/repos?per_page=100&page=#{page}"
-      end
+    cond do
+      options[:org] ->
+        "/orgs/#{owner(options)}/repos?per_page=100&page=#{page}"
+
+      options[:user] ->
+        "/users/#{owner(options)}/repos?per_page=100&page=#{page}"
+    end
+  end
+
+  @doc """
+    `rename_master_to_main/2` renames the given repo's default branch from master to main.
+    should fail silently.
+    API Doc: https://docs.github.com/en/rest/branches/branches#rename-a-branch
+    Ref: https://openssl.medium.com/renaming-the-default-branch-from-master-to-main-on-github-2f965cdbbc19
+  """
+  def rename_master_to_main(options, repo) do
+    "/repos/#{owner(options)}/#{repo}/branches/master/rename"
+    |> Github.post!(Poison.encode!(%{new_name: "main"}))
+    |> dbg()
```

Comment:
Do you want to keep the `debug` in the code?

Reply:
yeah, want to debug while it's running to get some insight into errors so can attempt to fix them. 💭